src/Sarif.Converters/HdfConverter.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    public class HdfConverter : ToolFileConverterBase
+    {
+        private const string PeriodString = ".";
+
+        private const string ToolInformationUri = "https://saf.cms.gov/#/normalize";
+
+        public override string ToolName => "Heimdall Tools";
+
+        public override void Convert(Stream input, IResultLogWriter output, OptionallyEmittedData dataToInsert)
+        {
+            input = input ?? throw new ArgumentNullException(nameof(input));
+            output = output ?? throw new ArgumentNullException(nameof(output));
+
+            var textReader = new StreamReader(input);
+            string jsonString = textReader.ReadToEnd();
+            var hdfFile = HdfFile.FromJson(jsonString);
+
+            (List<ReportingDescriptor>, List<Result>) rulesAndResults = ExtractRulesAndResults(hdfFile);
+
+            var run = new Run
+            {
+                Tool = new Tool
+                {
+                    Driver = new ToolComponent
+                    {
+                        Name = hdfFile.Platform.Name,
+                        Version = hdfFile.Version,
+                        InformationUri = new Uri(ToolInformationUri),
+                        Rules = rulesAndResults.Item1,
+                        SupportedTaxonomies = new List<ToolComponentReference>() { new ToolComponentReference() { Name = "NIST SP800-53 v5", Guid = "AAFBAB93-5201-419E-8443-D4925C542398" } }
+                    }
+                },
+                ExternalPropertyFileReferences = new ExternalPropertyFileReferences()
+                {
+                    Taxonomies = new List<ExternalPropertyFileReference>()
+                    {
+                        new ExternalPropertyFileReference()
+                        {
+                            Location = new ArtifactLocation()
+                            {
+                                Uri = new Uri("https://raw.githubusercontent.com/sarif-standard/taxonomies/main/NIST_SP800-53_v5.sarif"),
+                            },
+                            Guid = "AAFBAB93-5201-419E-8443-D4925C542398"
+                        }
+                    }
+                },
+                Results = rulesAndResults.Item2,
+            };
+
+            PersistResults(output, rulesAndResults.Item2, run);
+        }
+
+        internal new static void PersistResults(IResultLogWriter output, IList<Result> results, Run run)
```

Comment:
>internal new static [](http://example.com/codeflow?start=8&length=19)

do we need new static?

Reply:
was not with the static keyword, Visual Studio show this in the message section that this can be make static, so I just use that recommended fix

Reply:
make private static

Reply:
fixed to private new static. 
new is there because the base.

Reply:
i think you don't need this, since this is in the base and its the same thing. so, remove this and test.

Reply:
got it, let me try.

Reply:
works great, I have removed!